.github/workflows/documentation.yml

```diff
@@ -0,0 +1,26 @@
+name: Docs
+on: [push, pull_request, workflow_dispatch]
```

Comment:
so you are building docs for all pushes/PRs, but only publishing for pushes to main?  what does the `with publish_branch: gh-pages` do?

Reply:
1. To make sure that the documentation generation isn't broken I think it makes sense to always generate the docs event if they aren't published
2. `gh-pages` is the name of the branch that github points to for github pages.